File: images/normal_node/normal_node/modules/frr/frr_manager.go
package frr

import (
	"bytes"
	"fmt"
	"os"
	"os/exec"
)

func CopyFrrConfigurationFile() {
	containerName := os.Getenv("CONTAINER_NAME")

	sourceFilePath := fmt.Sprintf("/configuration/%s/route/frr.conf", containerName)

	destFilePath := "/etc/frr/frr.conf"

	cmd := exec.Command("cp", sourceFilePath, destFilePath)
	var out bytes.Buffer
	cmd.Stdout = &out

	err := cmd.Run()
	if err != nil {
		fmt.Println(err)
	}
}

func StartFrr() {
	cmd := exec.Command("service", "frr", "start")
	output, err := cmd.CombinedOutput()
	if err != nil {
		panic(err)
	}
	fmt.Println(string(output))
}

func Start() {
	enableFrr := os.Getenv("ENABLE_FRR")
	if enableFrr == "true" {
		fmt.Println("start frr")
		CopyFrrConfigurationFile()
		StartFrr()
	} else {
		fmt.Println("not start frr")
	}

}

File: images/malicious_node/malicious_node/modules/frr/frr_manager.py
import os


def start_frr():
    """
    启动 frr
    :return:
    """
    enableFrr = os.getenv("ENABLE_FRR")
    if enableFrr == "true":
        containerName = os.getenv("CONTAINER_NAME")
        sourceFilePath = f"/configuration/{containerName}/route/frr.conf"
        destFilePath = f"/etc/frr/frr.conf"
        copy_command = f"cp {sourceFilePath} {destFilePath}"
        start_frr_command = "service frr start"
        os.system(copy_command)
        os.system(start_frr_command)
        print("start frr", flush=True)
    else:
        print("not start frr", flush=True)
